lib/posix-fdtab/include/uk/posix-fdtab.h

```diff
@@ -0,0 +1,93 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+/* POSIX compatible file descriptor table */
+
+#ifndef __UK_POSIX_FDTAB_H__
+#define __UK_POSIX_FDTAB_H__ 1
+
+#include <uk/config.h>
+#include <uk/ofile.h>
+
+/* Internal fd type; should safely cast to what userspace expects an fd to be */
+typedef int uk_fd;
```

Comment:
That's fair; maybe there is a conceptual split to be made between `ukfdtab` - what implements most internal API & ofile management and whatnot, and `posix-fdtab` which provides syscalls, which would warrant this typedef. But then again, this split is so tenuous that I don't think it's worth bringing about right now.
Let's hope we won't ever have to either, because unlike this fix it's not a simple search&replace.

Reply:
Makes sense from where the idea is coming from. I am naively assume atm that we will not need to replace it, just because of the posix interface and I think it simplifies the code base if we hop less with data types that are effectively the same.